src/extensionsIntegrated/CustomSnippets/snippetsState.js

```diff
@@ -19,45 +19,81 @@
      */
     function loadSnippetsFromState() {
-        try {
-            const savedSnippets = prefs.get("snippetsList");
-            if (Array.isArray(savedSnippets)) {
-                // clear existing snippets and load from saved state
-                Global.SnippetHintsList.length = 0;
-                Global.SnippetHintsList.push(...savedSnippets);
-            }
-        } catch (e) {
-            console.error("something went wrong when trying to load custom snippets from preferences:", e);
-        }
+        return new Promise((resolve, reject) => {
+            const file = FileSystem.getFileForPath(SNIPPETS_FILE_PATH);
+
+            // true is for bypassCache, to get the latest content always
+            const readPromise = FileUtils.readAsText(file, true);
+
+            readPromise
+                .done(function (text) {
+                    try {
+                        const data = JSON.parse(text);
+                        if (data && data.snippets && Array.isArray(data.snippets)) {
+                            Global.SnippetHintsList = data.snippets;
+                        } else {
+                            // no snippets are present
+                            Global.SnippetHintsList = [];
+                        }
+                        resolve();
+                    } catch (error) {
+                        console.error("Error parsing snippets JSON:", error);
+                        Global.SnippetHintsList = []; // fallback
+                        resolve();
+                    }
+                })
+                .fail(function (error) {
+                    if (error === FileSystemError.NOT_FOUND) {
+                        // file is not present, empty array
+                        Global.SnippetHintsList = [];
+                        resolve();
+                    } else {
+                        console.error("Unable to load snippets from file storage:", error);
+                        Global.SnippetHintsList = [];
+                        reject(error);
+                    }
+                });
+        });
     }
 
     /**
-     * Save snippets to preferences
-     * This is called whenever snippets are modified
+     * this function is responsible to save snippets to file storage
+     * @returns {Promise} a promise that resolves when snippets are saved
      */
     function saveSnippetsToState() {
-        try {
-            prefs.set("snippetsList", [...Global.SnippetHintsList]);
-        } catch (e) {
-            console.error("something went wrong when saving custom snippets to preferences:", e);
-        }
+        return new Promise((resolve, reject) => {
+            const dataToSave = {
+                snippets: Global.SnippetHintsList
+            };
+
+            const file = FileSystem.getFileForPath(SNIPPETS_FILE_PATH);
+            // 2 is for pretty print
+            const jsonText = JSON.stringify(dataToSave, null, 2);
```

Comment:
Do not pretty print for machine only files, just doing `stringify(data)`  would be enough here. Pretty only files that will be read by humans in the editor.